components/FMenu.vue

```diff
@@ -211,13 +228,13 @@ const style = computed(
   })
 );
 
-const selectedOption = useVModelProxy<string | number | boolean>(props);
+const selectedOption = useVModelProxy<unknown>(props);
 
 const optionRefs = ref<HTMLElement[]>([]);
 
 const emit = defineEmits<{
-  (name: 'update:modelValue', value: string | number | null): void;
-  (name: 'select-option', value: string | number | null): void;
+  (name: 'update:modelValue', value: unknown): void;
+  (name: 'select-option', value: unknown): void;
   (name: 'toggle', value: boolean): void;
 }>();
```

Comment:
Can you put the defineEmits earlier in the file, after the props definition and defineExpose ?